core/evancore.vue.libs/src/components/permissions-editor/permissions-editor.ts

```diff
@@ -151,15 +164,23 @@ class PermissionsEditor extends mixins(EvanComponent) {
    *  - contractId: the id of the contract which was changed
    *  - permissions: the updated permissions object
    */
-  updateContractPermissions({contractId, permissions}) {
+  updateContractPermissions({ contractId, permissions }): void {
     this.containersPermissions[contractId].permissions = permissions;
-    this.permissionsChanged = !deepEqual(this.containersPermissions, this.initialPermissions);
+    this.permissionsChanged = !deepEqual(
+      this.containersPermissions,
+      this.initialPermissions,
+    );
+    /**
+     * Inform parent about dirty status of form
+     * TODO apply event listener to other listeners too
+     */
+    this.$emit('permissionsChanged', this.permissionsChanged);
```

Comment:
or put it into vuex UI state ?

Reply:
Let's not abuse it any further until we cleaned up there :p